frontend/lib/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -239,16 +330,23 @@ export class ArrowVegaLiteChart extends PureComponent<
     }
 
     if (!data || data.data.numRows === 0) {
-      const view = this.vegaView as any
-      // eslint-disable-next-line no-underscore-dangle
-      const viewHasDataWithName = view._runtime.data.hasOwnProperty(name)
-      if (viewHasDataWithName) {
+      // The new data is empty, so we remove the dataset from the
+      // chart view if the named dataset exists.
+      let namedDatasetExists = true
+      try {
+        this.vegaView.data(name)
+      } catch (e) {
+        namedDatasetExists = false
+      }
+
+      if (namedDatasetExists) {
         this.vegaView.remove(name, vega.truthy)
       }
       return
```

Comment:
I find it surprising that we cannot simply do

```typescript
try {
  this.vegaView.remove(name, vega.truthy)
} finally {
return
}
```

or similar

Reply:
I think that will probably also work. I changed it, but I'm not really sure when this is actually used... so hard to really test it in action. Probably used when you first use a chart with data, and then you call the chart with no data. But it doesn't sound like a reasonable use case.